draft-rescorla-tls-esni.md

```diff
@@ -154,6 +154,24 @@ innocuous domains, but is also providing protection for some private
 domains. Note that the backend server can be an unmodified TLS 1.3
 server.
 
+~~~~
+                +--------------------+       +---------------------+
+                |                    |       |                     |
+                |   2001:DB8::1111   |       |   2001:DB8::EEEE    |
+Client <------->|                    |<----->|                     |
+                | public.example.com |       | private.example.com |
+                |                    |       |                     |
+                +--------------------+       +---------------------+
+                  Client-Facing Server            Backend Server
+~~~~
+{: #proxy-mode title="Proxy Mode Topology"}
+
+In Proxy Mode, as in Split Mode, the provider is *not* the origin server
+for private domains. The difference from Split Mode is that the provider's
+server accepts and decrypts connections on behalf of the origin server. The
+provider's server then establishes a new connection to the origin server.
+The provider does have access to the plaintext of the connection. For most
+practical purposes, this can be treated as Split Mode.
```

Comment:
I think this creates confusion. From the TLS perspective, the Client-Facing server is the origin server.

Reply:
Thinking about it some more, the difference between these two modes is termination of the TLS connection, rather than where the origin server (and data) actually are. I will try to come up with something.